recipes/libuv/all/patches/1.38.0/fix-cmake.patch

```diff
@@ -0,0 +1,78 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 0496d36a..90615d57 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -298,13 +298,17 @@ if(APPLE OR CMAKE_SYSTEM_NAME MATCHES "DragonFly|FreeBSD|Linux|NetBSD|OpenBSD")
+   list(APPEND uv_test_libraries util)
+ endif()
+
+-add_library(uv SHARED ${uv_sources})
+-target_compile_definitions(uv
+-  INTERFACE
+-    USING_UV_SHARED=1
+-  PRIVATE
+-    BUILDING_UV_SHARED=1
+-    ${uv_defines})
++add_library(uv ${uv_sources})
++get_target_property(target_type uv TYPE)
++if (target_type STREQUAL "SHARED_LIBRARY")
++  target_compile_definitions(uv
++    INTERFACE
++      USING_UV_SHARED=1
```

Comment:
do you need to export USING_UV_SHARED as compiler definition?

Reply:
Hmm.  The test package seems to build without it, but it seems like a good idea to fully emulate the `uv` target's interface in Conan.  I'll add that to the Conanfile.

Reply:
`USING_UV_SHARED=1` added to `cpp_info.defines` in 965ebbd8d5eb7021ba15907868b440644c9a85b9.